### PR TITLE
Add customCloseIcon prop

### PR DESCRIPTION
**Pull request by moonboots**
@majapw @ljharb 

Adds a `customCloseIcon` prop to allow overriding the default `<CloseButton />`. I matched the behavior of `customArrowIcon`: default prop of null, fallback to original component inside `render`, and a wrapping css class.

### CODE REVIEW

File: src/components/DateRangePicker.jsx [hunk 1]
```diff
@@ -273,7 +277,9 @@ export default class DateRangePicker extends React.Component {
             <span className="screen-reader-only">
               {this.props.phrases.closeDatePicker}
             </span>
-            <CloseButton />
+            <div className="DateRangePickerInput__close">
```
Comment (majapw) [inline src/components/DateRangePicker.jsx]: This class name is wrong. You're not inside the input.

File: stories/DateRangePicker_input.js [hunk 2]
```diff
@@ -30,6 +30,19 @@ const TestCustomArrowIcon = () => (
...
+const TestCustomCloseIcon = () => (
+  <span
+    style={{
+      border: '1px solid #dce0e0',
+      backgroundColor: '#fff',
+      color: '#484848',
+      padding: '3px',
+    }}
+  >
+    {'X'}
```
Comment (majapw) [inline stories/DateRangePicker_input.js]: This can just be `X` no?

File: test/components/DateRangePickerInput_spec.jsx [hunk 3]
```diff
@@ -117,6 +117,17 @@ describe('DateRangePickerInput', () => {
...
   });
 
+  describe('props.customCloseIcon is a React Element', () => {
+    it('has custom icon', () => {
+      const wrapper = shallow(
+        <DateRangePickerInput
+          showClearDates
+          customCloseIcon={<span className="custom-close-icon" />}
+        />);
+      expect(wrapper.find('.DateRangePickerInput__calendar-icon .custom-close-icon'));
```
Comment (majapw) [inline test/components/DateRangePickerInput_spec.jsx]: This should fail... why would it be under `.DateRangePickerInput__calendar-icon`? Why isn't this failing?

**Comment by moonboots** [comment]
@majapw I've added a corresponding spec for SingleDatePicker. The spec you pointed out wasn't failing because it accidentally didn't have any assertions, only a dangling `expect(..);`, fixed now.

File: test/components/SingleDatePickerInput_spec.jsx [hunk 4]
```diff
@@ -54,6 +54,17 @@ describe('SingleDatePickerInput', () => {
...
     });
+
+    describe('props.customCloseIcon is a React Element', () => {
+      it('has custom icon', () => {
+        const wrapper = shallow(
+          <SingleDatePickerInput
+            showClearDates
+            customCloseIcon={<span className="custom-close-icon" />}
+          />);
+        expect(wrapper.find('.DateRangePickerInput__calendar-icon .custom-close-icon'));
```
Comment (majapw) [inline test/components/SingleDatePickerInput_spec.jsx]: You now have the same problem here in this spec...

Comment (moonboots) [inline test/components/SingleDatePickerInput_spec.jsx]: Oops, fixed

**Comment by zeemawn** [comment]
Correct me if I'm wrong, but shouldn't this whole thing rather be called "custom CLEAR icon"?
The 'X' might look like a close-icon, but its function is to clear the dates, isn't it?
(Besides that, it would be nice to have an actual close-button)

**Comment by ljharb** [comment]
It closes the datepicker, it doesn't clear the dates.

**Comment by majapw** [comment]
It actually does both... We overload the icon in use, so uh, I get the Q

**Comment by ljharb** [comment]
Lol well there you go

**Comment by ljharb** [comment]
@majapw can we unoverload them?

**Comment by interactivejohn** [comment]
I ran into the same issue as @zeemawn. I spent a good couple hours thinking that I could add a custom close icon to the default state, only to find out that it was only implemented for the `showClearDate` and `withFullScreenPortal` props. Maybe something can be added to the README that explains where the `customCloseIcon` is implemented? Meanwhile I will fork this package and start working on implementing a close functionality for the default `DatePicker` as requested above (I also need it for a project).

**Pull request merged**